src-extras/Database/LSMTree/Extras/Generators.hs

```diff
@@ -116,11 +116,19 @@ instance Arbitrary1 Monoidal.Update where
     Monoidal.Mupsert v -> Monoidal.Insert v : map Monoidal.Mupsert (shrinkVal v)
     Monoidal.Delete    -> []
 
-instance Arbitrary k => Arbitrary (Range k) where
-  arbitrary = oneof
-    [ FromToExcluding <$> arbitrary <*> arbitrary
-    , FromToIncluding <$> arbitrary <*> arbitrary
-    ]
+instance (Arbitrary k, Ord k) => Arbitrary (Range k) where
+  arbitrary = do
+    key1 <- arbitrary
+    key2 <- arbitrary `QC.suchThat` (/= key1)
```

Comment:
```suggestion
    key2 <- arbitrary
```
It's probably okay this way. It also means that for `k` that have few inhabitants, we don't get stuck. It's fine if we "degrade" into the unit range

Reply:
I don't see us running this with key types with few inhabitants, so it doesn't matter much either way. It just made the descriptions of the distribution a bit cleaner, because I can say `lb < ub` instead of having the `==` case leaking into everything.